### PR TITLE
Guard conditions

### PR DESCRIPTION
**Pull request by fsat**
- [x] Display error message if `CONDUCTR_HOST` or `CONDUCTR_IP` is specified as empty.
- [x] Fail if CLI is being invoked with unsupported Python version

### CODE REVIEW

**Comment by fsat** [comment]
Marked as wip - manual test to be done.

File: conductr_cli/host.py [hunk 1]
```diff
@@ -5,16 +5,20 @@
 from conductr_cli.docker import DockerVmType
 
 
+CONDUCTR_HOST = 'CONDUCTR_HOST'
+CONDUCTR_IP = 'CONDUCTR_HOST'
```
Comment (huntc) [inline conductr_cli/host.py]: Should that be `CONDUCTR_IP`?

Comment (fsat) [reply conductr_cli/host.py]: Yes - it should. Thank you for picking this up.

I need to add the test for `host.py`

**Review by huntc — APPROVED** [review]

**Comment by fsat** [comment]
**Manual test setup**

Install the local CLI changes.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ pip3 install -e .
Obtaining file:///Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli
Requirement already satisfied (use --upgrade to upgrade): requests>=2.6.0 in /usr/local/lib/python3.5/site-packages (from conductr-cli==0.46)
Requirement already satisfied (use --upgrade to upgrade): requests-toolbelt>=0.7.0 in /usr/local/lib/python3.5/site-packages (from conductr-cli==0.46)
Requirement already satisfied (use --upgrade to upgrade): argcomplete>=0.8.1 in /usr/local/lib/python3.5/site-packages (from conductr-cli==0.46)
Requirement already satisfied (use --upgrade to upgrade): pyhocon==0.2.1 in /usr/local/lib/python3.5/site-packages (from conductr-cli==0.46)
Requirement already satisfied (use --upgrade to upgrade): arrow>=0.6.0 in /usr/local/lib/python3.5/site-packages (from conductr-cli==0.46)
Requirement already satisfied (use --upgrade to upgrade): jsonschema==2.4 in /usr/local/lib/python3.5/site-packages (from conductr-cli==0.46)
Requirement already satisfied (use --upgrade to upgrade): pager<4.0,>=3.3 in /usr/local/lib/python3.5/site-packages (from conductr-cli==0.46)
Requirement already satisfied (use --upgrade to upgrade): prettytable<1.0,>=0.7 in /usr/local/lib/python3.5/site-packages (from conductr-cli==0.46)
Requirement already satisfied (use --upgrade to upgrade): pygments<3.0,>=2.0 in /usr/local/lib/python3.5/site-packages (from conductr-cli==0.46)
Requirement already satisfied (use --upgrade to upgrade): six<2.0,>=1.9 in /usr/local/lib/python3.5/site-packages (from conductr-cli==0.46)
Requirement already satisfied (use --upgrade to upgrade): toml<1.0,>=0.9 in /usr/local/lib/python3.5/site-packages (from conductr-cli==0.46)
Requirement already satisfied (use --upgrade to upgrade): pyparsing==2.0.3 in /usr/local/lib/python3.5/site-packages (from pyhocon==0.2.1->conductr-cli==0.46)
Requirement already satisfied (use --upgrade to upgrade): python-dateutil in /usr/local/lib/python3.5/site-packages (from arrow>=0.6.0->conductr-cli==0.46)
Installing collected packages: conductr-cli
  Running setup.py develop for conductr-cli
Successfully installed conductr-cli-0.46
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

**Guard against empty environment variable**

Run a ConductR sandbox.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ sandbox run 1.1.11
Starting ConductR nodes..
Starting container cond-0..
578998fc2a333d224985bbdd89c8bc23bf3053432220ba72c28874b86cb2ab3a
Waiting for ConductR to start...
ConductR has been started. Check current bundle status with: conduct info
```

Confirm `conduct info` works as is.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ conduct info
ID       NAME    #REP  #STR  #RUN
5a79de9  eslite     1     0     1

Felixs-MBP-2:conductr-cli felixsatyaputra$ conduct info --host 192.168.99.100
ID       NAME    #REP  #STR  #RUN
5a79de9  eslite     1     0     1

Felixs-MBP-2:conductr-cli felixsatyaputra$ conduct info --ip 192.168.99.100
ID       NAME    #REP  #STR  #RUN
5a79de9  eslite     1     0     1
```

Confirm error message is displayed if `CONDUCTR_HOST` is empty.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ export CONDUCTR_HOST=

Felixs-MBP-2:conductr-cli felixsatyaputra$ conduct info
Error: ConductR host address is not specified
Error: Please ensure either `CONDUCTR_HOST` environment is specified, or specify the ConductR host using `--host` argument

Felixs-MBP-2:conductr-cli felixsatyaputra$ env | sort | grep -i cond
CONDUCTR_HOST=
PWD=/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli
```

If `CONDUCTR_HOST` is specified, command works as expected.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ export CONDUCTR_HOST=192.168.99.100

Felixs-MBP-2:conductr-cli felixsatyaputra$ conduct info
ID       NAME    #REP  #STR  #RUN
5a79de9  eslite     1     0     1

Felixs-MBP-2:conductr-cli felixsatyaputra$ env | sort | grep -i cond
CONDUCTR_HOST=192.168.99.100
PWD=/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli
```

Similarly with `CONDUCTR_IP`.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ export CONDUCTR_IP=

Felixs-MBP-2:conductr-cli felixsatyaputra$ conduct info
Error: ConductR host address is not specified
Error: Please ensure either `CONDUCTR_HOST` environment is specified, or specify the ConductR host using `--host` argument

Felixs-MBP-2:conductr-cli felixsatyaputra$ env | sort | grep -i cond
CONDUCTR_IP=
PWD=/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli

Felixs-MBP-2:conductr-cli felixsatyaputra$ export CONDUCTR_IP=192.168.99.100

Felixs-MBP-2:conductr-cli felixsatyaputra$ conduct info
ID       NAME    #REP  #STR  #RUN
5a79de9  eslite     1     0     1

Felixs-MBP-2:conductr-cli felixsatyaputra$ env | sort | grep -i cond
CONDUCTR_IP=192.168.99.100
PWD=/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli
Felixs-MBP-2:conductr-cli felixsatyaputra$
```

**Guard against unsupported python version**

Use an unsupported python to run the CLI comands. Error message is prompted to user.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ python2.7 -m conductr_cli.conduct info
Unable to start CLI.
Current python version is 2.7.10
Please use python version 3.4 and above.

Felixs-MBP-2:conductr-cli felixsatyaputra$ python2.7 -m conductr_cli.sandbox run 1.1.11
Unable to start CLI.
Current python version is 2.7.10
Please use python version 3.4 and above.

Felixs-MBP-2:conductr-cli felixsatyaputra$ python2.7 -m conductr_cli.shazar ~/tmp/cd-test-config/
Unable to start CLI.
Current python version is 2.7.10
Please use python version 3.4 and above.
```

**Comment by fsat** [comment]
[ConductR CLI: ensure CONDUCTR_IP environment is not empty](https://trello.com/c/QbQFAUvf/407-conductr-cli-ensure-conductr-ip-environment-is-not-empty)

**Comment by fsat** [comment]
Fixes #196

**Pull request merged**